### PR TITLE
Lily/rationalized clean

### PR DESCRIPTION
**Pull request by lhz1029**
small pr to merge onto the doc rep pr before merging into master

### CODE REVIEW

File: enso/experiment/rationalized.py [hunk 1]
```diff
@@ -211,14 +214,10 @@ def fit(self, X, Y):
...
-        rationale_texts = [
-            rationale['text'] 
-            for doc in rationales 
-            for rationale in doc
-        ]
-        docs = np.asarray([self.NLP(str(x), disable=['ner', 'tagger', 'textcat']) for x in X])
-        rationale_docs = np.asarray([self.NLP(rationale) for rationale in rationale_texts if len(rationale)])
-        self._train_rationale_model(docs, rationale_docs)
+        rationale_texts = [rationale["text"] for doc in rationales for rationale in doc]
+        docs = np.asarray([self.NLP(str(x), disable=["ner", "tagger", "textcat"]) for x in X])
```
Comment (benleetownsend) [inline enso/experiment/rationalized.py:218]: can we do this with NLP.pipe?

File: enso/experiment/rationalized.py [hunk 2]
```diff
@@ -264,21 +266,71 @@ def _featurize(self, doc):
...
+            for label, rationale_vecs in rationale_vecs_by_class.items()
+        }
+        self.normalized_rationale_proto_by_class = OrderedDict({
+            label: rationale_proto / np.linalg.norm(rationale_proto)
+            for label, rationale_proto in rationale_proto_by_class.items()
+        })
 
-    def _train_rationale_model(self, docs, rationale_docs):
+    def _rationale_weight(self, word, rationale_proto):
+        cosine_sim = np.dot(word.vector / np.linalg.norm(word.vector), rationale_proto)
```
Comment (benleetownsend) [inline enso/experiment/rationalized.py:291]: By using cosine similarity we give a strong negative weight to anything that differs a lot, would a cosine distance not make more sense.

File: enso/experiment/rationalized.py (hunk 1 above)
Comment (lhz1029) [reply enso/experiment/rationalized.py:218]: ooh good call! just tested and it does give a speedup in both training and predict. Will change in enso now, not sure about how that translates into production when we get to it.

File: enso/experiment/rationalized.py (hunk 2 above)
Comment (lhz1029) [reply enso/experiment/rationalized.py:291]: that's a reasonable hypothesis for sure! However, I tested and it seems to perform less well (closer to the baseline). My intuition is that the negative weights end up creating even more of an emphasis on the rationale vector direction.

**Pull request merged**